### PR TITLE
add machineSet ami filters

### PR DESCRIPTION
**Pull request by enxebre**
Add ami filters so the actuator can get the right image

### CODE REVIEW

**Comment by enxebre** [comment]
cc @bison

**Comment by bison** [comment]
@enxebre: Link to the test output is fixed. The expected output in the tests needs to be updated.

File: machines/aws/machine-set.yaml [hunk 1]
```diff
@@ -28,6 +28,19 @@ spec:
           kind: AWSMachineProviderConfig
           ami:
             id: {{.AWS.Image}}
```
Comment (bison) [inline machines/aws/machine-set.yaml]: What happens if you pass both here? Which does it choose? This seems like it should be optional.

**Pull request merged**